pkg/tools/etcd_tools.go

```diff
@@ -418,20 +418,31 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 	}
 }
 
-func checkEtcd(host string) error {
+// GetEtcdVersion performs a version check against the provided Etcd server, returning a triplet
+// of the release version, internal version, and error (if any).
+func GetEtcdVersion(host string) (relVersion string, intVersion string, e error) {
 	response, err := http.Get(host + "/version")
 	if err != nil {
-		return err
+		return "", "", err
 	}
 	defer response.Body.Close()
+
 	body, err := ioutil.ReadAll(response.Body)
 	if err != nil {
-		return err
+		return "", "", err
 	}
-	if !strings.HasPrefix(string(body), "etcd") {
-		return fmt.Errorf("unknown server: %s", string(body))
+
+	var dat map[string]interface{}
+	if err := json.Unmarshal(body, &dat); err != nil {
+		return "", "", fmt.Errorf("unknown server: %s", string(body))
 	}
-	return nil
+	if dat["releaseVersion"] != nil {
```

Comment:
In the context of this code, the original version should be safe (I think, since it's unmarshalling a json). But to be extra safe, made the changes.